### PR TITLE
First Draft Workspace Trust Editor

### PR DESCRIPTION
**Pull request by sbatten**
This PR implements the first draft of a 1st Class UI editor for Trusted Workspace State

### CODE REVIEW

**Comment by sbatten** [comment]
as this is hidden behind the feature flag, we will continue dev in `main`

File: src/vs/workbench/contrib/workspace/browser/workspaceTrustEditor.css [hunk 1]
```diff
@@ -0,0 +1,73 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+.monaco-icon-label.file-icon.workspacetrusteditor-name-file-icon.ext-file-icon.tab-label::before {
+	font-family: 'codicon';
+	content: '\eb53';
```
Comment (roblourens) [inline src/vs/workbench/contrib/workspace/browser/workspaceTrustEditor.css:8]: Isn't there a codicon class you are supposed to use instead of this?

Comment (sbatten) [reply src/vs/workbench/contrib/workspace/browser/workspaceTrustEditor.css:8]: editor tabs are controlled by the file icon theme and not codicons... cc @misolori

Comment (misolori) [reply src/vs/workbench/contrib/workspace/browser/workspaceTrustEditor.css:8]: We've had requests to allow codicons in the editor tabs cc @aeschli

**Pull request merged**